src/defaultsettings.cpp

```diff
@@ -26,9 +26,71 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "mapgen/mapgen.h" // Mapgen::setDefaultSettings
 #include "util/string.h"
 
+
+/* based on https://github.com/systemd/systemd/blob/7aed43437175623e0f3ae8b071bbc500c13ce893/src/hostname/hostnamed.c#L406
+ * this could be done in future with D-Bus using query:
+ * busctl get-property org.freedesktop.hostname1 /org/freedesktop/hostname1 org.freedesktop.hostname1 Chassis
+ */
+bool detect_touch()
+{
+#if defined(__ANDROID__)
+	return true;
+#else
+	std::string chassis_type;
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
+
+	// device-tree platforms (non-X86)
+	std::ifstream dtb_file("/proc/device-tree/chassis-type");
+	if (dtb_file.is_open()) {
+		std::getline(dtb_file, chassis_type);
+		chassis_type.pop_back();
```

Comment:
what does this remove and could it be better solved by trimming the string of spacing?

Reply:
the kernel returns `\0` at the end of the string which we don't care about, so we always strip it, imho this is the best solution as it's always there.